File: AsalSayıSorg.py
def Asalmi(x:int):
    flag = True
    if x == 2:
        return flag
    elif x % 2== 0:
        flag = False
    
    if flag:
        for i in range(3,int(x/2)):
            if x % i ==0:
                print(x,"sayısının",i,"sayısına bölümü:",x/i,"kalan ise 0'dır.")
                flag = False
                break
    return flag

x = int(input("Sayı girin : "))
print(Asalmi(x))

File: AsalSayıSorg.cs
using System;

class Program
{
    static bool Asalmi(int x)
    {
        bool flag = true;

        if (x == 2)
        {
            return flag;
        }
        else if (x % 2 == 0)
        {
            flag = false;
        }

        if (flag)
        {
            for (int i = 3; i < x / 2; ++i)
            {
                if (x % i == 0)
                {
                    Console.WriteLine($"{x} sayısının {i} sayısına bölümü: {Convert.ToDouble(x) / i}, kalan ise 0'dır.");
                    flag = false;
                    break;
                }
            }
        }

        return flag;
    }

    static void Main()
    {
        Console.Write("Sayı girin: ");
        int x = Convert.ToInt32(Console.ReadLine());

        Console.WriteLine(Asalmi(x));
    }
}
